### PR TITLE
Expose Unicode.Char.Parser

### PR DESCRIPTION
**Pull request by rnjtranjan**

### CODE REVIEW

File: core/src/Streamly/Unicode/Char/Parser.hs [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    , lower
+    , upper
+    , mark
+    , print
+    , punctuation
+    , separator
+    , space
+    , symbol
+
+    -- digits
```
Comment (adithyaov) [inline core/src/Streamly/Unicode/Char/Parser.hs:42]: Make this a heading?

**Comment by adithyaov** [comment]
@harendra-kumar 
1. Do we want to expose all the APIs?
2. We need to add more docs if we are exposing it. Maybe some examples.

**Review by harendra-kumar — CHANGES_REQUESTED** [review]
As discussed, rename the module to Streamly.Unicode.Parser.

**Review by harendra-kumar — CHANGES_REQUESTED** [review]
Change "print" to "printable".

File: core/src/Streamly/Unicode/Parser.hs [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    , separator
+    , space
+    , symbol
+
+    -- * Digits
+    , digit
+    , octDigit
+    , hexDigit
+
+    -- * Numeric
```
Comment (adithyaov) [inline core/src/Streamly/Unicode/Parser.hs:47]: This should be `Numbers`

File: core/src/Streamly/Unicode/Parser.hs [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    , signed
+    , numeric
```
Comment (adithyaov) [inline core/src/Streamly/Unicode/Parser.hs:49]: This should be under `Digit`

**Review by adithyaov — COMMENTED** [review]
Please address the comments.

**Comment by adithyaov** [comment]
Fixed the comments.

File: core/src/Streamly/Internal/Unicode/Char/Parser.hs [hunk 4]
```diff
@@ -77,7 +76,7 @@ import qualified Streamly.Internal.Data.Parser as Parser
 
 -- XXX It may be possible to implement faster predicates for ASCII byte stream.
 -- We can measure if there is a signficant difference and if so we can add such
--- predicates to Streamly.Unicode.Char.Parser.Latin1.
+-- predicates to Streamly.Unicode.Parser.Latin1.
```
Comment (adithyaov) [inline core/src/Streamly/Internal/Unicode/Char/Parser.hs:79]: What is `Unicode.Parser.Latin1`? Where should `Latin1` stuff go in? @harendra-kumar

**Pull request merged**